### PR TITLE
fix azure spn hunter

### PR DESCRIPTION
**Pull request by itaysk**
Fixes https://github.com/aquasecurity/kube-hunter/issues/370

### CODE REVIEW

File: kube_hunter/modules/hunting/aks.py [hunk 1]
```diff
@@ -46,11 +46,16 @@ def get_key_container(self):
             logger.debug("failed getting pod info")
         else:
             pods_data = r.json().get("items", [])
+            suspecious_volume_names = []
```
Comment (evrardjp) [inline kube_hunter/modules/hunting/aks.py:49]: nit: isn't this spelled suspicious?

File: kube_hunter/modules/hunting/aks.py [hunk 2]
```diff
@@ -46,11 +46,16 @@ def get_key_container(self):
...
             for pod_data in pods_data:
+                for volume in pod_data["spec"]["volumes"]:
```
Comment (evrardjp) [inline kube_hunter/modules/hunting/aks.py:51]: nit: Not sure whether the spec can _not_ have volumes. If that's the case, isn't it worth writing as `for volume in pod_data['spec'].get('volumes',[])` ? Or maybe have a try: except: KeyError?

File: kube_hunter/modules/hunting/aks.py (hunk 1 above)
Comment (itaysk) [reply kube_hunter/modules/hunting/aks.py]: yes thanks

File: kube_hunter/modules/hunting/aks.py (hunk 2 above)
Comment (itaysk) [reply kube_hunter/modules/hunting/aks.py]: you are absolutely right I forgot to check for failure cases. Thank you for spotting that.

**Review by danielsagi — APPROVED** [review]
lgtm

**Pull request merged**